### PR TITLE
Cleaning the whole repo! +remove singletons!

### PR DESCRIPTION
**Pull request by sim0504**
… one commit. bang! Happy? Don't be sad :(

### CODE REVIEW

File: Util/constant.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 # TeamColor
 class TeamColor(Enum):
 
-    def from_str(self, color: str):
+    def from_str(self):
```
Comment (sim0504) [inline Util/constant.py]: I just realize, this is pretty stupid...

File: ai/GameDomainObjects/ShittyField.py [hunk 2]
```diff
@@ -30,7 +31,8 @@ def __init__(self, protobuf_line):
         self.thickness = protobuf_line.thickness
 
 
-class Shitty_Field:
+# noinspection PyPep8,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyAttributeOutsideInit,PyMethodMayBeStatic,PyMethodMayBeStatic,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences
```
Comment (sim0504) [inline ai/GameDomainObjects/ShittyField.py]: Wow thanks PyCharm!

File: Engine/filters/ball_kalman_filter.py [hunk 3]
```diff
@@ -28,6 +27,7 @@ def observation_model(self):
         return np.array([[1, 0, 0, 0],   # Position x
                          [0, 0, 1, 0]])  # Position y
 
+    # noinspection PyPep8Naming
```
Comment (PhiBabin) [inline Engine/filters/ball_kalman_filter.py]: another

**Comment by sim0504** [comment]
I have modifiy the Position a bit to avoid type collision with np.ndarray. If you need the array form of the position, use pos.array.

File: Engine/filters/robot_kalman_filter.py [hunk 4]
```diff
@@ -45,11 +44,13 @@ def control_input_model(self):
                          [0,  0,  0],  # Position Theta
                          [0,  0, dt]])  # Speed Theta
 
+    # noinspection PyPep8,PyPep8Naming
```
Comment (PhiBabin) [inline Engine/filters/robot_kalman_filter.py]: another

File: Engine/filters/robot_kalman_filter.py [hunk 5]
```diff
@@ -60,15 +61,9 @@ def process_covariance(self):
 
         return G
 
+    # noinspection PyPep8
     def observation_covariance(self):
```
Comment (PhiBabin) [inline Engine/filters/robot_kalman_filter.py]: another

File: Util/ai_command.py [hunk 6]
```diff
@@ -15,7 +15,7 @@
                                     'ball_collision,'
                                     'pathfinder_on')
 
-
+# noinspection PyPep8Naming
```
Comment (PhiBabin) [inline Util/ai_command.py]: another

File: Util/role.py [hunk 7]
```diff
@@ -8,3 +9,8 @@ class Role(IntEnum):
     SECOND_DEFENCE = 3
     FIRST_ATTACK = 4
     SECOND_ATTACK = 0
+
+    # noinspection PyTypeChecker
```
Comment (PhiBabin) [inline Util/role.py]: another

File: ai/Algorithm/evaluation_module.py [hunk 8]
```diff
@@ -23,6 +23,7 @@ def player_with_ball(min_dist_from_ball=1.2*ROBOT_RADIUS, our_team=None):
         return None
 
 
+# noinspection PyUnusedLocal
```
Comment (PhiBabin) [inline ai/Algorithm/evaluation_module.py]: another

File: ai/GameDomainObjects/ShittyField.py (hunk 2 above)
Comment (PhiBabin) [reply ai/GameDomainObjects/ShittyField.py]: ShittyField is too shitty for Pycharm

File: ai/STA/Strategy/penalty_defense.py [hunk 9]
```diff
@@ -9,6 +9,7 @@
 from ai.states.game_state import GameState
 
 
+# noinspection PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences
```
Comment (PhiBabin) [inline ai/STA/Strategy/penalty_defense.py]: another

File: ai/STA/Strategy/prepare_kickoff_defense.py [hunk 10]
```diff
@@ -12,6 +12,7 @@
 from ai.states.game_state import GameState
 
 
+# noinspection PyArgumentList,PyArgumentList,PyArgumentList,PyArgumentList,PyArgumentList,PyArgumentList,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences,PyUnresolvedReferences
```
Comment (PhiBabin) [inline ai/STA/Strategy/prepare_kickoff_defense.py]: wtf

File: ai/STA/Strategy/strategy.py [hunk 11]
```diff
@@ -75,7 +79,7 @@ def get_current_state(self) -> List[Tuple[int, str, str, str]]:
             except AttributeError:
                 tactic_name = "DEFAULT"
             state.append((current_tactic.player, str(current_tactic)+" "+current_tactic.status_flag.name+" " +
-                          current_tactic.current_state.__name__, str(current_tactic), current_tactic.target))
+                          tactic_name, str(current_tactic), current_tactic.target))
```
Comment (PhiBabin) [inline ai/STA/Strategy/strategy.py]: `current_tactic.current_state.__name__` name of the state inside of the tactic.

File: Engine/Communication/receiver/vision_receiver.py [hunk 12]
```diff
@@ -1,7 +1,11 @@
 # Under MIT License, see LICENSE.txt
 
 from ipaddress import ip_address
-from socket import socket, AF_INET, SOCK_DGRAM, IPPROTO_IP, IP_ADD_MEMBERSHIP, inet_aton, INADDR_ANY, timeout, SOL_SOCKET, SO_REUSEADDR
+
+# noinspection PyUnresolvedReferences
+from socket import AF_INET, SOCK_DGRAM, IPPROTO_IP, IP_ADD_MEMBERSHIP, INADDR_ANY, SOL_SOCKET, SO_REUSEADDR
+# noinspection PyUnresolvedReferences
+from socket import inet_aton, socket, timeout
```
[the remark this answers is not in the document]
Comment (sim0504) [reply Engine/Communication/receiver/vision_receiver.py]: Je suis pas sur pourquoi il arrive pas à trouver les flags

File: ai/STA/Strategy/prepare_kickoff_defense.py (hunk 10 above)
Comment (sim0504) [reply ai/STA/Strategy/prepare_kickoff_defense.py]: ouais par defaut il en met un par erreur, c'est stupide

**Comment by sim0504** [comment]
I have remove the no inpection in the main files. only the «shitty» file contains them to avoid warning when inspectiing code. less noisy, more fun.

**Pull request merged**